### ISSUE TITLE
About the code

### ISSUE THREAD

**Issue by Z1zs** [zepingyu0512/llava-mechanism#2]
Thanks for your amazing work. I checked the file `modeling_llava.py` but didn't found a hook to record the hidden states of llm.  
I think you may want to provide a intrusive method in the forward function of ` language_model ` in `LlavaForConditionalGeneration` rather than itself? Or have I missed something？

And maybe there is better practice to register a hook in a non-intrusive way :)

**Comment by zepingyu0512** [comment]
Hi, I've uploaded the modeling_llama.py file. Llava is fine-tuned based on Llama2.

And thanks. Updating the code is on my to-do list. :)

**Comment by Z1zs** [comment]
Thanks very much for the reply, I have been able to reproduce the results!

**Comment by qm-intel** [comment]
@Z1zs Could you please explain how you were able to run the code and produce the results? I am getting the error, [explained here](https://github.com/zepingyu0512/llava-mechanism/issues/3#issue-2881625262)

and the outputs are `odict_keys(['logits', 'past_key_values', 'image_hidden_states'])` , what is the input in this line?

```
all_pos_layer_input, all_pos_attn_output, all_pos_residual_output, all_pos_ffn_output, all_pos_layer_output, \
all_last_attn_subvalues, all_pos_coefficient_scores, all_attn_scores = transfer_output(outputs[2])
```
If the output[2], which is `image_hidden_states`, it raises the error mentioned above. Can you guide what you exactly did?

**Comment by Z1zs** [comment]
@qm-intel Hi there. I had also met this question before, which I thought should be related with Transformers version. You can try to downgrade the version of transformers, but I don't remember the exact version I used for reproducing. Old version allows to get hidden states as well as attention score during `model.generate()`.
